### PR TITLE
[DIGI-10959]: Nucleus native objects

### PR DESCRIPTION
**Pull request by WBertan**
- [x] Use Kotlin 1.3.0.

- [x] Using moshi to handle parsing objects to and from json format, also using moshi annotations because it can handle the default values in the data classes.

- [x] Add classes for native objects having as reference the existing models in the main Digi.me app model module.

- [x] Write sort of test framework for the models classes where run a set of tests converting an object to json and a json to object, try to keep it easy to extend/grow and be reusable (especially jsons representing nested structures).

- [x] Add extension for Moshi.Builder to add custom models adapters.

- [x] Self-review.

- [x] Ready for review.

### CODE REVIEW

File: digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        val container: String?,
+
+        @Json(name = "id")
+        val id: String?,
+
+        @Json(name = "entityid")
+        val entityId: String?,
+
+        /**
+         * jfs account id, which is converted to [Account.id]
```
Comment (hmaclean82) [inline digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt]: @WBertan let's remove these sort of comments, they're not relevant for SDK users and even if we integrate this module with digime app, these comments are not very helpful...

File: digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+         */
+        @Json(name = "accountentityid")
+        val accountEntityId: String?,
+
+        /**
+         * mapped to [ContentItem.text2] together with [.currency]
```
Comment (hmaclean82) [inline digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt]: @WBertan definitely let's remove these "mapped to ContentItem.." comments

File: digime-models/src/main/java/me/digi/sdk/models/health_and_fitness/DaySummaryActiveMinutes.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2009-2018 digi.me Limited. All rights reserved.
+ */
+
+package me.digi.sdk.models.health_and_fitness
+
+import com.squareup.moshi.Json
+import me.digi.sdk.models.ItemDetails
+
+data class DaySummaryActiveMinutes(
```
Comment (hmaclean82) [inline digime-models/src/main/java/me/digi/sdk/models/health_and_fitness/DaySummaryActiveMinutes.kt]: @WBertan as this is extracted from the top level `DailyActivitySummary` it's not so useful to include just the model without including the extraction logic contained in `populate`. Or do you think we should be packaging that somewhere else?

File: digime-models/src/main/java/me/digi/sdk/models/health_and_fitness/EpochSummary.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2009-2018 digi.me Limited. All rights reserved.
+ */
+
+package me.digi.sdk.models.health_and_fitness
+
+import com.squareup.moshi.Json
+import me.digi.sdk.models.ItemDetails
+
+data class EpochSummary(
```
Comment (hmaclean82) [inline digime-models/src/main/java/me/digi/sdk/models/health_and_fitness/EpochSummary.kt]: @WBertan `EpochSummary, StressSummary, BodyCompositionSummary, MoveIqSummary` are not synced so I think it's best just to remove these models

Comment (WBertan) [reply digime-models/src/main/java/me/digi/sdk/models/health_and_fitness/EpochSummary.kt]: Done.

File: digime-models/src/main/java/me/digi/sdk/models/health_and_fitness/DaySummaryActiveMinutes.kt (hunk 3 above)
Comment (WBertan) [reply digime-models/src/main/java/me/digi/sdk/models/health_and_fitness/DaySummaryActiveMinutes.kt]: Done, removed the `DaySummary...` classes, tests and unused components.

File: digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt (hunk 1 above)
Comment (WBertan) [reply digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt]: Done.

File: digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt (hunk 2 above)
Comment (WBertan) [reply digime-models/src/main/java/me/digi/sdk/models/finance/Transaction.kt]: Done.

**Pull request merged**